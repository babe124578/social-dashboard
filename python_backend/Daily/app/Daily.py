from flask_cors import CORS, cross_origin
from flask import Flask
import io
import pandas as pd
import boto3

app = Flask(__name__)

cors = CORS(app)
app.config['CORS_HEADERS'] = 'Content-Type'


def get_key(field):
    df = pd.read_csv('credentials.csv')
    return df[field][0]


def get_dataframe():
    s3 = boto3.client('s3',
                      aws_access_key_id=get_key('Access key ID'),
                      aws_secret_access_key=get_key('Secret access key'))
    obj = s3.get_object(Bucket='csv-storage', Key='time.csv')
    df = pd.read_csv(io.BytesIO(obj['Body'].read()))
    return df


def daily_user_each_day():
    dailyCount = dict()

    for i in get_dataframe()['0']:
        if i.split(' ')[0] not in dailyCount:
            dailyCount[i.split(' ')[0]] = 1
        else:
            dailyCount[i.split(' ')[0]] += 1

    return [list(dailyCount.keys()), list(dailyCount.values())]


@cross_origin()
@app.route('/test')
def test():
    return 'hellotest'


@app.route('/Daily')
def daily():
    return {'labels': daily_user_each_day()}


if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5001, use_reloader=True)


# run docker run
