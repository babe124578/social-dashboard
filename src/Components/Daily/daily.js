import React, { Component } from "react";
import { Line } from "react-chartjs-2";

class daily extends Component {
    state = {
        options: {
            responsive: true,
            title: {
                display: true,
                text: 'Daily post on social media'
            },
            tooltips: {
                mode: 'label'
            },
            hover: {
                mode: 'dataset'
            },
            scales: {
                xAxes: [{ display: true, scaleLabel: { show: true, labelString: 'Month' } }],
                yAxes: [{ display: true, scaleLabel: { show: true, labelString: 'Value' }, ticks: { suggestedMin: -10, suggestedMax: 250 } }]
            }
        },
        datas: {
            labels: null
        }
    }
    componentDidMount() {
        fetch('https://dashdaily2.ap-northeast-1.elasticbeanstalk.com/Daily')
            .then(res => res.json())
            .then((data) => {
                this.setState({
                    datas: {
                        labels: data.labels[0],
                        datasets: [{
                            label: "date_count",
                            data: data.labels[1],
                            hoverBackgroundColor: '#B8F8EC',
                            pointBorderColor: '#FF0000',
                            pointBackgroundColor: '#FF0000',
                            pointBorderWidth: 1,
                            hoverBorderDash: [5,5],
                            lineTension: '0',
                            radius: 2,
                            pointRadius: 2.5,
                            pointHoverRadius: 3
                        }]
                    }
                })
            })
    }
    render() {
        return (
            <div>
                <Line data={this.state.datas} options={this.state.options}></Line>
            </div>
        )
    }
}

export default daily;