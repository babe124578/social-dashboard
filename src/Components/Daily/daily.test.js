import React from 'react';
import Daily from "./daily.js";
import { shallow } from 'enzyme';
import shallowRenderer from 'react-test-renderer/shallow'

it('should render without crashing', () => {
    shallow(<Daily />);
});

it('shallow snapshot', () => {
    const renderer = new shallowRenderer()
    const snapshot = renderer.render(<Daily />)
    expect(snapshot).toMatchSnapshot()
});


