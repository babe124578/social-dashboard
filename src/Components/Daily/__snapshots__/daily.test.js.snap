// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow snapshot 1`] = `
<div>
  <Line
    data={
      Object {
        "labels": null,
      }
    }
    options={
      Object {
        "hover": Object {
          "mode": "dataset",
        },
        "responsive": true,
        "scales": Object {
          "xAxes": Array [
            Object {
              "display": true,
              "scaleLabel": Object {
                "labelString": "Month",
                "show": true,
              },
            },
          ],
          "yAxes": Array [
            Object {
              "display": true,
              "scaleLabel": Object {
                "labelString": "Value",
                "show": true,
              },
              "ticks": Object {
                "suggestedMax": 250,
                "suggestedMin": -10,
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "Daily post on social media",
        },
        "tooltips": Object {
          "mode": "label",
        },
      }
    }
  />
</div>
`;
